corehq/apps/geospatial/tasks.py

```diff
@@ -14,3 +28,42 @@ def geo_cases_reassignment_update_owners(domain, case_owner_updates_dict, task_k
     finally:
         celery_task_tracker = CeleryTaskTracker(task_key)
         celery_task_tracker.mark_completed()
+
+
+@task(queue='geospatial_queue', ignore_result=True)
+def index_es_docs_with_location_props(domain):
+    celery_task_tracker = get_celery_task_tracker(domain, INDEX_ES_TASK_HELPER_BASE_KEY)
+    if celery_task_tracker.is_active():
+        return
```

Comment:
Does this ignore a task if another one is already running?
If yes, I believe, ideally, we should do that check before accepting the task, assuming that is possible here.
For how its written currently, HQ would ignore the request to index docs silently.

Reply:
It ignores the task only if another task is already running for a specific domain. This won't block other domains from starting tasks.

We could move this check to the `_handle_geospatial_es_index()` save function on the toggle. That way, we will only start a new task if the Celery task tracker reports that one isn't already running.

Reply:
Addressed in [d0d4d2b](https://github.com/dimagi/commcare-hq/pull/35126/commits/d0d4d2b395bb1a332b2533f10302be270632df05), but let me know if you think a different approach would be better.